File: timeplan/front/src/app/services/group.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Group} from "../models/group";
import {Student} from "../models/student";
import {Subject} from "../models/subject";


@Injectable({
  providedIn: 'root'
})

export class GroupService{

  constructor(private http: HttpClient){}

  createGroup(name:string, students: Student[]):Observable<Group>{
    return this.http.post<Group>('/api/gp/create/'+name, students);
  }

  addSubject(idgroup:number, subjects: Subject[]):Observable<Group>{
    return this.http.post<Group>('/api/gp/addsub/'+idgroup, subjects);
  }

  removeSubjectFromGroup(idgroup:number, idsub:number):Observable<Group>{
    return this.http.get<Group>('/api/gp/remove/'+idgroup+"/"+idsub);
  }

  getAllGroups():Observable<Group[]>{
    return this.http.get<Group[]>('/api/gp/getgp');
  }

  deleteById(idgroup:number):Observable<void>{
    return this.http.delete<void>('/api/gp/delete/'+idgroup);
  }

}

File: timeplan/fapi/src/main/java/com/netcracker/fapi/service/GroupService.java
package com.netcracker.fapi.service;

import com.netcracker.fapi.model.Group;
import com.netcracker.fapi.model.Student;
import com.netcracker.fapi.model.Subject;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;

import java.util.*;

@Service
public class GroupService {

    @Value("${backend.server.url}")
    private String backendServerUrl;
    public Group createGroup(String name, Student[] students){
        HashMap<String,List<Student>> hashMap = new HashMap<>();
        hashMap.put(name, Arrays.asList(students));
        RestTemplate restTemplate = new RestTemplate();
        return restTemplate.postForEntity(backendServerUrl + "/api/group/create", hashMap, Group.class).getBody();
    }

    public List<Group> getAllGroups(){
        RestTemplate restTemplate = new RestTemplate();
        Group [] groups = restTemplate.getForObject(backendServerUrl + "/api/group/getgroups", Group[].class);
        return groups == null ? Collections.emptyList(): Arrays.asList(groups);
    }

    public void removeSubjectFromGroup(Long idgroup, Long idsub){
        RestTemplate restTemplate = new RestTemplate();
        restTemplate.getForObject(backendServerUrl + "/api/group/remove/"+idgroup+"/"+idsub,void.class);
    }

    public Group addSubjects(Long idgrop, Subject[] subjects){
        HashMap<Long , List<Subject>> hashMap = new HashMap<>();
        hashMap.put(idgrop,Arrays.asList(subjects));
        RestTemplate restTemplate = new RestTemplate();
        return restTemplate.postForEntity(backendServerUrl + "/api/group/addsubjects",hashMap,Group.class).getBody();
    }

    public void deleteGroupById(Long id){
        RestTemplate restTemplate = new RestTemplate();
        restTemplate.delete(backendServerUrl + "/api/group/delete/"+id);
    }
}
